actions/lint/action.yml

```diff
@@ -0,0 +1,122 @@
+author: hoverkraft
+branding:
+  icon: check-circle
+  color: blue
+
+inputs:
+  working-directory:
+    description: |
+      Working directory where lint commands are executed.
+      Can be absolute or relative to the repository root.
+    required: false
+    default: "."
+  container:
+    description: "Whether running in container mode (skips checkout and node setup)"
+    required: false
+    default: "false"
+  report-file:
+    description: |
+      Path to lint report file to process as GitHub annotations.
+      Supports common formats like checkstyle XML, eslint JSON, etc.
+      If not specified, no report processing is done.
+    required: false
+    default: ""
+  fail-on-error:
+    description: "Whether to fail the action if linting errors are found"
+    required: false
+    default: "true"
+
+runs:
+  using: "composite"
+  steps:
+    - shell: bash
+      # FIXME: workaround until will be merged: https://github.com/actions/runner/pull/1684
+      run: mkdir -p ./self-lint-action/ && cp -r $GITHUB_ACTION_PATH/../* ./self-lint-action/
+
+    - id: setup-node
+      if: inputs.container != 'true'
+      uses: ./self-lint-action/setup-node
+      with:
+        working-directory: ${{ inputs.working-directory }}
+        dependencies-cache: |
+          nx
+          prettier
+
+    - id: get-package-manager
+      if: inputs.container == 'true'
+      uses: ./self-lint-action/get-package-manager
+      with:
+        working-directory: ${{ inputs.working-directory }}
+
+    - shell: bash
+      id: run-lint
+      working-directory: ${{ inputs.working-directory }}
+      env:
+        RUN_SCRIPT_COMMAND: ${{ inputs.container == 'true' && steps.get-package-manager.outputs.run-script-command || steps.setup-node.outputs.run-script-command }}
+        FAIL_ON_ERROR: ${{ inputs.fail-on-error }}
+      run: |
+        set +e  # Don't exit on error, we want to process the report
+
+        echo "👕 Running lint..."
+        $RUN_SCRIPT_COMMAND lint
+        LINT_EXIT_CODE=$?
+
+        echo "lint-exit-code=$LINT_EXIT_CODE" >> $GITHUB_OUTPUT
+
+        if [ "$FAIL_ON_ERROR" = "true" ] && [ $LINT_EXIT_CODE -ne 0 ]; then
+          echo "::error::Linting failed with exit code $LINT_EXIT_CODE"
+        fi
+
+        # Always exit successfully here; we'll handle the failure later if needed
+        exit 0
+
+    - shell: bash
+      if: inputs.report-file != ''
+      working-directory: ${{ inputs.working-directory }}
+      env:
+        REPORT_FILE: ${{ inputs.report-file }}
+      run: |
+        set -e
+
+        if [ ! -f "$REPORT_FILE" ]; then
+          echo "::warning::Lint report file not found: $REPORT_FILE"
+          exit 0
+        fi
+
+        echo "📊 Processing lint report: $REPORT_FILE"
+
+        # Detect report format and process accordingly
+        if echo "$REPORT_FILE" | grep -q "\.json$"; then
+          echo "Detected JSON format"
+          # Process ESLint JSON format
```

Comment:
You can us ethis action to process eslint report file: https://github.com/ataylorme/eslint-annotate-action